### PR TITLE
Add max_stepsize on fixed rank

### PR DESCRIPTION
**Pull request by mateuszbaran**
Since we have problems with finding the right local `injectivity_radius` on `FixedRankMatrices`, I think it would be good to at least get a reasonable `max_stepsize` based on what Matlab Manopt has.

### CODE REVIEW

**Comment by kellertuer** [comment]
Maybe that Matlab/Manopt reference is a bit weak and could be a link if we find a good way to link it? Besides that this looks good.

**Comment by kellertuer** [comment]
Sure, we can also keep it. Just that the reference is a bit vague.

**Comment by mateuszbaran** [comment]
How would you like me to give that reference? A link like this: https://github.com/NicolasBoumal/manopt/blob/97b6eb6b185334ab7b3991585ed2c044d69ee905/manopt/manifolds/fixedrank/fixedrankembeddedfactory.m#L76-L78 ?

**Comment by kellertuer** [comment]
Yes.

File: ext/ManoptManifoldsExt/manifold_functions.jl [hunk 1]
```diff
@@ -10,6 +10,17 @@ function max_stepsize(M::TangentBundle, p)
     return max_stepsize(M.manifold, p[M, :point])
 end
 
+"""
+    max_stepsize(M::FixedRankMatrices, p)
+
+Return a reasonable guess of maximum step size on `FixedRankMatrices` following
+the choice of typical distance in Matlab Manopt, i.e. dimension of `M`. See
+https://github.com/NicolasBoumal/manopt/blob/97b6eb6b185334ab7b3991585ed2c044d69ee905/manopt/manifolds/fixedrank/fixedrankembeddedfactory.m#L76-L78
```
Comment (kellertuer) [inline ext/ManoptManifoldsExt/manifold_functions.jl:18]:
```suggestion
[this note](https://github.com/NicolasBoumal/manopt/blob/97b6eb6b185334ab7b3991585ed2c044d69ee905/manopt/manifolds/fixedrank/fixedrankembeddedfactory.m#L76-L78)
```

**Review by kellertuer — APPROVED** [review]
Is this a good idea for a new version or shall we wait for a next larger feature?

Besides that
* Please add an entry to the changeling
* maybe bump version.

**Comment by mateuszbaran** [comment]
I've added changelog entry and bumped version.

**Pull request merged**